src/modules/layouts/PageContext.js

```diff
@@ -28,14 +34,62 @@ const PageDataProvider = ({ children, value }) => {
           absolutePath
         }
       }
+
+      #IMPORT t() translated UI JSON here.
```

Comment:
Ok, so this is all beyond cool. Big fan of both the code and the level of documentation and explanation you have provided here. Am I right in assuming that, whenever we want to add a new locale it's as easy as creating a new key here?

Reply:
@andytudhope Yep that's one hundred percent correct. It's my least favorite part only because it **HAS** to be devils advocate. The elephant in the room with this integration is how we have to bring in our json files using graphQL.

Every time a new string is added to the UI.json the dev has to update this graphql query for each locale it's added to (not very elegant, but necessary). Another solution to importing the data could be to have some js file with a bunch of JSON imports at the top, but then we wouldn't have the validation checks GraphQL offers us (especially in the case a UI.json doesn't exist for a specific locale). 

 Since all UI based translations are expected to be integrated by a developer first (without the expectation of them being able to translate a UI element) the workflow would be:
1. Developer adds UI element with inlined text (let's say an action like "Cancel") 
2. Developer adds "Cancel": "Cancel" to UI.json.
3. (easiest) Developer goes to GrapiQL (___graphql route)  and builds simple query that auto generates copy and past query for them. 
4. Developer copies and pastes updated query in PageContext.js